packages/dashboard/app/withdraw/utils/handlers.ts

```diff
@@ -0,0 +1,245 @@
+    }
+  }
+};
+
+export const handleRefreshBalances = async (
+  ethAddress: string | undefined,
+  setLoading: (loading: boolean) => void,
+  setBalances: (balances: TokenBalance[]) => void,
+  showStatus: ShowStatusFn,
+  showError: ShowErrorFn
+) => {
+  if (!ethAddress) return;
+
+  try {
+    setLoading(true);
+    const ethBalance = await fetchEthBalance(ethAddress);
+    const erc20TokenBalances = await fetchERC20TokenBalances(ethAddress);
+
+    const balances = [ethBalance.balances[0],...erc20TokenBalances.balances];
+
+    if (erc20TokenBalances.success && ethBalance.success) {
+      setBalances(balances);
+      showStatus('Token balances fetched successfully', 'success');
+    } else {
+      showStatus('Failed to fetch all token balances', 'warning');
+      if (erc20TokenBalances.error) {
+        showError(erc20TokenBalances.error, 'Token Balance Error');
+      }
+      if (ethBalance.error) {
+        showError(ethBalance.error, 'Token Balance Error');
+      }
+    }
+  } catch (err: any) {
+    showStatus('Failed to fetch token balances', 'error');
+    showError('Error fetching token balances', 'Error');
+  } finally {
+    setLoading(false);
+  }
+};
+
+export const handleSubmit = async (
+  e: React.FormEvent,
+  selectedToken: TokenBalance | null,
+  withdrawAmount: string,
+  withdrawAddress: string,
+  agentPKP: IRelayPKP,
+  sessionSigs: SessionSigs,
+  setSubmitting: (submitting: boolean) => void,
+  setWithdrawAmount: (amount: string) => void,
+  setWithdrawAddress: (address: string) => void,
+  showStatus: ShowStatusFn,
+  showError: ShowErrorFn,
+  refreshBalances: () => void
+) => {
+  e.preventDefault();
+
+  if (!selectedToken || !withdrawAmount || !withdrawAddress) {
+    showStatus('Please fill all fields', 'warning');
+    return;
+  }
+
+  if (parseFloat(withdrawAmount) === 0) {
+    showStatus('Withdrawal amount cannot be zero', 'warning');
+    return;
+  }
+
+  if (!ethers.utils.isAddress(withdrawAddress)) {
+    showStatus('Invalid withdrawal address', 'error');
+    return;
+  }
+
+  if (selectedToken) {
+    if (selectedToken.symbol === 'ETH') {
+      const parsedAmount = ethers.utils.parseUnits(withdrawAmount, selectedToken.decimals);
+      if (parsedAmount.gte(selectedToken.rawBalance)) {
+        showStatus(`You have entered an amount greater than or equal to your balance. Please enter a smaller amount.`, 'warning');
+        return;
+      }
+    }
+  }
+
+  try {
+    setSubmitting(true);
+    showStatus('Preparing withdrawal...', 'info');
+    litNodeClient = await initLitNodeClient();
+
+    const pkpWallet = new PKPEthersWallet({
+      pkpPubKey: agentPKP.publicKey,
+      litNodeClient: litNodeClient,
+      controllerSessionSigs: sessionSigs,
+      rpc: BASE_MAINNET_RPC
+    })
+
+    await pkpWallet.init();
+
+    const amount = ethers.utils.parseUnits(withdrawAmount, selectedToken.decimals);
+
+    let transactionResult;
+
+    if (selectedToken.symbol === 'ETH') {
```

Comment:
This might be a big refactor at this point and possibly better a follow up. But treating the native coin as an token pollutes the code with so much of these ifs that it is just not worth it

The benefit of merging them seems reasonable when thinking their display is similar (which really is not because we always display the native token) but they have so many logical differences (ones pays gas and is a native transfer the other can be maxed and triggers a contract) that make the tradeoff not worth it

Just wanted to write that. We don't need to fix it now as it is expensive, it works, we need this shipped and we are going to refactor a bunch of stuff already